src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/EETypeNode.cs

```diff
@@ -483,6 +491,25 @@ private void OutputNullableTypeParameter(NodeFactory factory, ref ObjectDataBuil
             }
         }
 
+        /// <summary>
+        /// Returns true if <paramref name="type"/> is an interface type that could be implemented by an array.
+        /// </summary>
+        private static bool IsGenericInterfaceImplementedByArray(TypeDesc type)
+        {
+            // As a quick test we can use the fact that interfaces implemented by array are all non-generic.
+            if (!type.IsInterface || type.Instantiation.Length != 1)
+                return false;
+
+            TypeDesc elementType = type.Instantiation[0];
+            foreach (var interfaceType in elementType.MakeArrayType().RuntimeInterfaces)
```

Comment:
This seems really quite expensive. How about making an array of the generic type definitions of runtime interfaces implemented by arrays and then doing a comparison there?